### PR TITLE
API fixes

### PR DESCRIPTION
**Pull request by Havivw**

### CODE REVIEW

File: surch/repo.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+#    * limitations under the License.
+import os
+import logging
+import subprocess
+from time import time
+
+import yaml
+import retrying
+from tinydb import TinyDB
+from . import logger
```
Comment (Havivw) [inline surch/repo.py]: one line down

File: surch/repo.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+# This string is a template for blob_url to redirect you
+# for the problematic commit.
+# It get organization, repository_name, sha, file_name
+BLOB_URL_TEMPLATE = 'https://github.com/{0}/{1}/blob/{2}/{3}'
+HOME_PATH = os.path.expanduser("~")
+DEFAULT_PATH = os.path.join(HOME_PATH, 'surch')
+LOG_PATH = os.path.join(HOME_PATH, 'problematic_commit.json')
```
Comment (Havivw) [inline surch/repo.py]: results.json

File: surch/repo.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        self.search_list = self._create_search_strings(list(search_list))
+        self.url = url
+        if not os.path.isdir(local_path):
+            os.makedirs(local_path)
+        self.local_path = local_path
+        self.quiet_git = '--quiet' if quiet_git else ''
+
+        lgr.setLevel(logging.DEBUG if verbose else logging.INFO)
+
+    @classmethod
```
Comment (Havivw) [inline surch/repo.py]: remove

File: surch/repo.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        for string in search_list:
+            search_strings = "{0} --or -e '{1}'".format(search_strings, string)
+        return search_strings
+
+    def _find_problematic_commits_in_directory(self, string_to_search,
+                                               directory=None,
+                                               repository_name=None,
+                                               organization_name=None):
+        """ Create list of all problematic commits"""
+        bad_files = []
```
Comment (Havivw) [inline surch/repo.py]: ?

File: surch/repo.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+        bad_files = []
+        self.repository_name = repository_name or self.repository_name
+        self.organization_name = organization_name or self.organization_name
+        self.directory = directory or self.full_path
+        lgr.info('Now scan the {0} repository'.format(self.repository_name))
+        for commit in self._get_all_commits_list(self.directory):
+            bad_files.append(self._get_bad_files(self.directory, commit,
```
Comment (Havivw) [inline surch/repo.py]: bad_files -> something else

File: surch/repo.py [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+        except subprocess.CalledProcessError:
+            return []
+
+    def _get_bad_files(self, directory, commit, string_to_search):
+        """ Run git grep"""
+        try:
+            bad_files = subprocess.check_output(
+                'git -C {0} grep -l -e {1} {2}'.format(directory,
+                                                       string_to_search,
+                                                       commit), shell=True)
```
Comment (Havivw) [inline surch/repo.py]: bad_files = subprocess.check_output(
                'git -C {0} grep -l -e {1} {2}'.format(
                        directory, string_to_search, commit), shell=True)

File: surch/repo.py [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+                    #  get them we do pass
+                    pass
+                user_name, user_mail = self._get_user_details(
+                    sha, self.directory)
+                self._write_dict_to_db(sha, file_name, user_name, user_mail,
+                                       blob_url)
+
+    def _get_user_details(self, sha, directory):
+        git_log_command = "git -C {0} show -s  {1} |grep 'Author:'|" \
+                          "cut -d':' -f2|sort|uniq"
```
Comment (Havivw) [inline surch/repo.py]: parse in python

File: surch/repo.py [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+        user_details = subprocess.check_output(
+            git_log_command.format(directory, sha), shell=True)
+        name = user_details.rsplit('<', 1)[0]
+        mail = (user_details.rsplit('<', 1)[1]).rsplit('>', 1)[0]
+        return name, mail
+
+    def _write_dict_to_db(self, sha, files_name,
```
Comment (Havivw) [inline surch/repo.py]: _write_result

File: surch/surch.py [hunk 9]
```diff
@@ -310,6 +32,33 @@ def main():
...
+              help='Git http url.')
+@click.option('-p', '--path', required=False, default=DEFAULT_PATH,
+              help='This path contain the repos clone.')
+@click.option('-l', '--log', required=False, default=LOG_PATH,
+              help='Log file for bad commits')
+@click.option('-v', '--verbose', default=False, is_flag=True)
+@click.option('-q', '--quiet', default=True, is_flag=False)
+def repo(search, url, log, path, verbose, quiet):
+    """check single repository
+    :param search:
```
Comment (Havivw) [inline surch/surch.py]: remove params

File: surch/surch.py [hunk 10]
```diff
@@ -310,6 +32,33 @@ def main():
...
+@click.option('-u', '--url', required=True,
+              help='Git http url.')
+@click.option('-p', '--path', required=False, default=DEFAULT_PATH,
+              help='This path contain the repos clone.')
+@click.option('-l', '--log', required=False, default=LOG_PATH,
+              help='Log file for bad commits')
+@click.option('-v', '--verbose', default=False, is_flag=True)
+@click.option('-q', '--quiet', default=True, is_flag=False)
+def repo(search, url, log, path, verbose, quiet):
+    """check single repository
```
Comment (Havivw) [inline surch/surch.py]: capitalize first letter

File: surch/organization.py [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import os
+import time
+import logging
+# import subprocess
+
+import yaml
+# import click
```
Comment (Havivw) [inline surch/organization.py]: בםצצקמאד?

File: surch/organization.py [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+# import retrying
+import requests
+from tinydb import TinyDB
```
Comment (Havivw) [inline surch/organization.py]: //

File: surch/organization.py [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+HOME_PATH = os.path.expanduser("~")
+DEFAULT_PATH = os.path.join(HOME_PATH, 'surch')
+LOG_PATH = os.path.join(HOME_PATH, 'problematic_commit.json')
+
+lgr = logger.init()
+
+
+
+
+def _calculate_performance_to_second(start, end):
```
Comment (Havivw) [inline surch/organization.py]: reuse from repo

File: surch/organization.py [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+        :type git_password: basestring
+        :param local_path: this path contain the repos clone
+        :type local_path: basestring
+        :param verbose: user verbose mode
+        :type verbose: bool
+        """
+        self.error_summary = []
+        self.git_user = git_user
+        self.db = log_path
+        self.search_list = self._create_search_strings(list(search_list))
```
Comment (Havivw) [inline surch/organization.py]: reuse all possible code from "repo"

File: surch/organization.py [hunk 15]
```diff
@@ -0,0 +1,221 @@
...
+        self.local_path = os.path.join(local_path, organization)
+        self.quiet_git = '--quiet' if quiet_git else ''
+        self.verbose = verbose
+
+        lgr.setLevel(logging.DEBUG if verbose else logging.INFO)
+
+    @classmethod
+    def from_config_file(cls, config_file, verbose=False, quiet_git=True):
+        """ Define vars from "config.yaml" file"""
+        conf_vars = utiles.from_config_file(config_file, verbose, quiet_git)
```
Comment (Havivw) [inline surch/organization.py]: utils

File: surch/organization.py [hunk 16]
```diff
@@ -0,0 +1,221 @@
...
+        if not os.path.isdir(local_path):
+            os.makedirs(local_path)
+        self.local_path = os.path.join(local_path, organization)
+        self.quiet_git = '--quiet' if quiet_git else ''
+        self.verbose = verbose
+
+        lgr.setLevel(logging.DEBUG if verbose else logging.INFO)
+
+    @classmethod
+    def from_config_file(cls, config_file, verbose=False, quiet_git=True):
```
Comment (Havivw) [inline surch/organization.py]: change name

File: surch/organization.py [hunk 17]
```diff
@@ -0,0 +1,221 @@
...
+    def _search(self):
+        start = time.time()
+        self.get_github_repo_list()
+        self._clone_repositorys()
+        self.search_in_commits()
+        total = repo._calculate_performance_to_second(start, time.time())
+        lgr.debug('Total time: {0} seconds'.format(total))
+        self._print_error_summary()
+
+
```
Comment (Havivw) [inline surch/organization.py]: api changes (skipped,git_user,git_password=default)

File: surch/organization.py [hunk 18]
```diff
@@ -0,0 +1,221 @@
...
+            search_strings = "{0} --or -e '{1}'".format(search_strings, string)
+        return search_strings
+
+    def _print_error_summary(self):
+        if self.error_summary:
+            lgr.info(
+                'Summary of all errors: \n{0}'.format(
+                    '\n'.join(self.error_summary)))
+
+    def _search(self):
```
Comment (Havivw) [inline surch/organization.py]: not_private

File: surch/organization.py [hunk 19]
```diff
@@ -0,0 +1,221 @@
...
+        start = time.time()
+        self.get_github_repo_list()
+        self._clone_repositorys()
```
Comment (Havivw) [inline surch/organization.py]: repositories

File: surch/organization.py [hunk 20]
```diff
@@ -0,0 +1,221 @@
...
+        self.search_in_commits()
+        total = repo._calculate_performance_to_second(start, time.time())
```
Comment (Havivw) [inline surch/organization.py]: total_time

Comment (Havivw) [inline surch/organization.py]: move to utils---> convert_to_seconds

File: surch/repo.py [hunk 21]
```diff
@@ -0,0 +1,254 @@
...
+
+    repo = Repo(search_list=search_list, url=url, local_path=local_path,
+                log_path=log_path, verbose=verbose, quiet_git=quiet_git)
+
+    repo._find_problematic_commits_in_directory(
+        directory=directory, repository_name=repository_name,
+        organization_name=organization_name)
+
+
+def repo_search(search_list, url, local_path=DEFAULT_PATH,
```
Comment (Havivw) [inline surch/repo.py]: search

Comment (Havivw) [inline surch/repo.py]: repo_url

File: surch/repo.py [hunk 22]
```diff
@@ -0,0 +1,254 @@
...
+        self._print_error_summary()
+        lgr.debug('Total time: {0} seconds'.format(total))
+
+
+def _calculate_performance_to_second(start, end):
+    """ Calculate the runnig time"""
+    return str(round(end - start, 3))
+
+
+def _clone_or_pull_repository(search_list, url, repository_name,
```
Comment (Havivw) [inline surch/repo.py]: change to public

File: surch/repo.py [hunk 23]
```diff
@@ -0,0 +1,254 @@
...
+                        'user_name': user_name,
+                        'user_mail': user_mail,
+                        'blob_url': blob_url})
+
+    def _print_error_summary(self):
+        if self.error_summary:
+            lgr.info('Summary of all errors: \n{0}'.format(
+                '\n'.join(self.error_summary)))
+
+    def _search(self):
```
Comment (Havivw) [inline surch/repo.py]: change to public

File: surch/surch.py [hunk 24]
```diff
@@ -13,367 +14,108 @@
...
-        lgr.debug('Total time: {0} seconds'.format(total))
-        self._print_error_summary()
+@repository.command(name='args')
+@click.argument('url', required=True)
+@click.option('-S', '--strings', required=True, multiple=True,
+              help='List of secrets you want to search.')
+@click.option('-p', '--path', required=False, default=DEFAULT_PATH,
+              help='This path contain the repos clone.')
+@click.option('-l', '--log', required=False, default=LOG_PATH,
+              help='Log file for bad commits')
```
Comment (Havivw) [inline surch/surch.py]: merge command

File: surch/organization.py [hunk 25]
```diff
@@ -0,0 +1,221 @@
...
+
+        lgr.setLevel(logging.DEBUG if verbose else logging.INFO)
+
+    @classmethod
+    def from_config_file(cls, config_file, verbose=False, quiet_git=True):
+        """ Define vars from "config.yaml" file"""
+        conf_vars = utiles.from_config_file(config_file, verbose, quiet_git)
+        return cls(**conf_vars)
+
+    def get_github_repo_list(
```
Comment (nir0s) [inline surch/organization.py]: all functions which are not part of the API should be private

File: surch/organization.py [hunk 26]
```diff
@@ -0,0 +1,231 @@
...
+        :type local_path: basestring
+        :param verbose: user verbose mode
+        :type verbose: bool
+        """
+        self.error_summary = []
+        self.git_user = git_user
+        self.db = log_path
+        self.search_list = search_list
+        self.ignore_repository = skipped_repo or []
+        if not organization or not git_user:
```
Comment (Havivw) [inline surch/organization.py]: git user and password

File: surch/repo.py [hunk 27]
```diff
@@ -0,0 +1,266 @@
...
+LOG_PATH = os.path.join(HOME_PATH, 'results.json')
+
+lgr = logger.init()
+
+
+class Repo(object):
+    def __init__(
+            self,
+            search_list,                #
+            repo_url,
```
Comment (Havivw) [inline surch/repo.py]: #

**Pull request merged**